digdag-standards/src/main/java/io/digdag/standards/command/kubernetes/KubernetesClientConfig.java

```diff
@@ -19,42 +19,50 @@ public static KubernetesClientConfig create(final Optional<String> name,
             final Config systemConfig,
             final Config requestConfig)
     {
-        if (requestConfig.has("kubernetes")) {
-            // from task request config
-            return KubernetesClientConfig.createFromTaskRequestConfig(name, requestConfig.getNested("kubernetes"));
-        }
-        else {
-            // from system config
-            return KubernetesClientConfig.createFromSystemConfig(name, systemConfig);
+
+        String clusterName = null;
+        if (name.isPresent()) {
+            clusterName = name.get();
         }
-    }
 
-    private static KubernetesClientConfig createFromTaskRequestConfig(final Optional<String> name,
-            final Config config)
-    {
-        // TODO
-        // We'd better to customize cluster config by task request config??
-        throw new UnsupportedOperationException("Not support yet");
-    }
+        Config extractedSystemConfig = null;
+        if (systemConfig != null && systemConfig.get("agent.command_executor.type", String.class, "").equals("kubernetes")) {
+            if (clusterName == null) {
+                clusterName = systemConfig.get(KUBERNETES_CLIENT_PARAMS_PREFIX + "name", String.class); // ConfigException
+            }
+            final String keyPrefix = KUBERNETES_CLIENT_PARAMS_PREFIX + clusterName + ".";
+            extractedSystemConfig = StorageManager.extractKeyPrefix(systemConfig, keyPrefix);
+        }
 
-    @VisibleForTesting
-    static KubernetesClientConfig createFromSystemConfig(final Optional<String> name,
-            final io.digdag.client.config.Config systemConfig)
-    {
-        final String clusterName;
-        if (!name.isPresent()) {
-            if (!systemConfig.get("agent.command_executor.type", String.class, "").equals("kubernetes")) {
-                throw new ConfigException("agent.command_executor.type: is not 'kubernetes'");
+        Config extractedRequestConfig = null;
+        if (requestConfig != null && requestConfig.has("kubernetes")) {
+            if (clusterName == null) {
+                clusterName = requestConfig.get("name", String.class); // ConfigException
             }
-            clusterName = systemConfig.get(KUBERNETES_CLIENT_PARAMS_PREFIX + "name", String.class);
+            extractedRequestConfig = requestConfig.getNested("kubernetes");
         }
-        else {
-            clusterName = name.get();
+
+        // Create a config that merges RequestConfig with SystemConfig
+        final Config config;
+        if (extractedSystemConfig != null && extractedRequestConfig != null) {
```

Comment:
If you set empty Config as default value in both `extractedSystemConfig` and `extractedRequestConfig`.
You can simply merge them without `if ... else if ...` ?
And at last check required keys are set.